packages/components/src/Select.tsx

```diff
@@ -1,50 +1,75 @@
 import React, { useCallback } from 'react';
 import classNames from 'classnames';
+import { useForwardedRef } from '@deephaven/react-hooks';
 
 export type SelectProps = {
   children: React.ReactNode;
   onBlur?: React.FocusEventHandler<HTMLSelectElement>;
   onChange: (value: string) => void;
+  onKeyDown?: React.KeyboardEventHandler<HTMLSelectElement>;
+  id?: string;
   className?: string;
   defaultValue?: string;
   name?: string;
+  title?: string;
   value?: string;
   disabled?: boolean;
   'data-testid'?: string;
+  'aria-label'?: string;
 };
 
-function Select({
-  children,
-  onBlur,
-  onChange,
-  className,
-  defaultValue,
-  name,
-  value,
-  disabled,
-  'data-testid': dataTestId,
-}: SelectProps): JSX.Element {
-  const handleChange = useCallback(
-    event => {
-      onChange(event.target.value);
-    },
-    [onChange]
-  );
+const Select = React.forwardRef<HTMLSelectElement, SelectProps>(
+  (props, forwardedRef) => {
+    const {
+      children,
+      onBlur,
+      onChange,
+      onKeyDown,
+      id,
+      className,
+      defaultValue,
+      name,
+      title,
+      value,
+      disabled,
+      'data-testid': dataTestId,
+      'aria-label': ariaLabel,
+      ...rest
+    } = props;
 
-  return (
-    <select
-      className={classNames('custom-select', className)}
-      onBlur={onBlur}
-      onChange={handleChange}
-      defaultValue={defaultValue}
-      value={value}
-      name={name}
-      disabled={disabled}
-      data-testid={dataTestId}
-    >
-      {children}
-    </select>
-  );
-}
+    const ref = useForwardedRef<HTMLSelectElement>(forwardedRef);
+
+    const handleChange = useCallback(
+      (event: React.ChangeEvent<HTMLSelectElement>): void => {
+        onChange(event.target.value);
+      },
+      [onChange]
+    );
+
+    return (
+      <select
+        id={id}
+        ref={ref}
+        className={classNames('custom-select', className)}
+        onBlur={onBlur}
+        onChange={handleChange}
+        onKeyDown={onKeyDown}
+        defaultValue={defaultValue}
+        value={value}
+        name={name}
+        title={title}
+        disabled={disabled}
+        data-testid={dataTestId}
+        aria-label={ariaLabel}
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...rest}
```

Comment:
If we are going to spread props, any reason not to include others? Seems like most of them could be included in `rest`

Reply:
Like what?

Reply:
Looks like this would work

```typescript
const {
  children,
  onChange,
  className,
  'data-testid': dataTestId,
  'aria-label': ariaLabel,
  ...rest
} = props;
```

```jsx
<select
  ref={ref}
  className={classNames('custom-select', className)}
  onChange={handleChange}
  data-testid={dataTestId}
  aria-label={ariaLabel}
  // eslint-disable-next-line react/jsx-props-no-spreading
  {...rest}
>
  {children}
</select>
```